pkg/operator/sync.go

```diff
@@ -244,6 +246,94 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.imgLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		} else {
+			for key, d := range cm.Data {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
```

Comment:
I guess what we're saying here is to check if the data is encoded, and if not, just use it regardless of format

I know we talked about this briefly already, could you remind me what format it is expected to be in? Is it expected that it sometimes is encoded, and sometimes not?

Reply:
not sure which format it is to be in but I know that the two (binary vs non binary) cannot overlap. Meaning there cannot be two keys that have the same name in either one. So I do not think decoding both does us any harm.